src/main/java/org/folio/rest/impl/RecordBulkAPI.java

```diff
@@ -28,33 +26,31 @@
 
   private static final Logger LOG = LogManager.getLogger();
 
-  private CQLWrapper getCQL(String query, String table) throws FieldException {
+  private CQLWrapper getCQL(String query, String table, int limit, int offset) throws FieldException {
     CQL2PgJSON cql2pgJson = new CQL2PgJSON(table + ".jsonb");
     return new CQLWrapper(cql2pgJson, query);
   }
 
   @Validate
   @Override
-  public void getRecordBulkIds(RecordBulkIdsGetField field,
-        RecordBulkIdsGetRecordType recordType, String query, String lang,
-        RoutingContext routingContext, Map<String, String> okapiHeaders,
-        Handler<AsyncResult<Response>> asyncResultHandler,
-        Context vertxContext) {
-
-    try {
-      if (recordType.toString().equalsIgnoreCase(HOLDING_TYPE)) {
-        CQLWrapper wrapper = getCQL(query, HOLDING_TABLE);
-        PgUtil.streamGet(HOLDING_TABLE, RecordID.class, wrapper, null,
-          "ids", routingContext, okapiHeaders, vertxContext);
-      } else {
-        CQLWrapper wrapper = getCQL(query, INSTANCE_TABLE);
-        PgUtil.streamGet(INSTANCE_TABLE, RecordID.class, wrapper, null,
-          "ids", routingContext, okapiHeaders, vertxContext);
+  public void getRecordBulkIds(RecordBulkIdsGetField field, RecordBulkIdsGetRecordType recordType, String query,
+                               int offset, int limit, String lang, RoutingContext routingContext,
+                               Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+                               Context vertxContext) {
+      try {
+        if (recordType.toString().equalsIgnoreCase(HOLDING_TYPE)) {
+          CQLWrapper wrapper = getCQL(query, HOLDING_TABLE, limit, offset);
+          PgUtil.streamGet(HOLDING_TABLE, RecordID.class, wrapper, null,
+            "ids", routingContext, okapiHeaders, vertxContext);
+        } else {
+          CQLWrapper wrapper = getCQL(query, INSTANCE_TABLE, limit, offset);
+          PgUtil.streamGet(INSTANCE_TABLE, RecordID.class, wrapper, null,
+            "ids", routingContext, okapiHeaders, vertxContext);
+        }
+      } catch (Exception e) {
+        LOG.error(e.getMessage(), e);
+        asyncResultHandler.handle(Future.succeededFuture(GetRecordBulkIdsResponse
+          .respond500WithTextPlain(e.getMessage())));
```

Comment:
Why has the formatting of this code been changed during this pull request?

If folks want to make significant changes to the formatting, it could be easier to follow if that was made in a separate pull request to the behavioural change.

Reply:
Updated to the previous formatting